src/mcu_opcodes.cpp

```diff
@@ -1254,6 +1294,15 @@ void MCU_Opcode_SHLR(uint8_t opcode, uint8_t opcode_reg)
         MCU_SetStatus(C, STATUS_C);
         MCU_SetStatusCommon(data, operand_size);
     }
+    else if (opcode_reg == 0x01 && operand_type != GENERAL_IMMEDIATE) // SHAR
+    {
+        uint32_t data = MCU_Operand_Read();
+        uint32_t C = data & 0x1;
+        data >>= 1;
```

Comment:
MSB (bit 7 in byte mode, bit 15 in word mode) should be preserved according to manual

Reply:
good catch! fixed, I hope the fix is more correct